src/braket/circuits/circuit.py

```diff
@@ -1217,16 +1229,73 @@ def _create_openqasm_header(
                 f"{serialization_properties.qubit_reference_type} supplied."
             )
 
-        frame_wf_declarations = self._generate_frame_wf_declarations()
+        frame_wf_declarations = self._generate_frame_wf_defcal_declarations(gate_calibrations)
         if frame_wf_declarations:
             ir_instructions.append(frame_wf_declarations)
         return ir_instructions
 
-    def _generate_frame_wf_declarations(self) -> Optional[str]:
-        frames = {}
-        waveforms = {}
+    def _validate_ngc_uniqueness(
+        self,
+        gate_calibrations: GateCalibrations,
+        frames: Dict[Frame],
+        waveforms: Dict[ArbitraryWaveform],
+    ) -> None:
+        for key, calibration in gate_calibrations.calibration_data.items():
+            for frame in calibration._frames.values():
+                _validate_uniqueness(frames, frame)
+                frames[frame.id] = frame
+            for waveform in calibration._waveforms.values():
+                _validate_uniqueness(waveforms, waveform)
+                waveforms[waveform.id] = waveform
```

Comment:
Could you please add in a comment on why we are validating uniqueness for the frames and wfs?